controllers/gitrepository_controller.go

```diff
@@ -219,18 +269,36 @@ func (r *GitRepositoryReconciler) reconcile(ctx context.Context, repository sour
 		git.CheckoutOptions{
 			GitImplementation: repository.Spec.GitImplementation,
 			RecurseSubmodules: repository.Spec.RecurseSubmodules,
-		})
+		},
+	)
 	if err != nil {
 		return sourcev1.GitRepositoryNotReady(repository, sourcev1.GitOperationFailedReason, err.Error()), err
 	}
 	commit, revision, err := checkoutStrategy.Checkout(ctx, tmpGit, repository.Spec.URL, auth)
 	if err != nil {
 		return sourcev1.GitRepositoryNotReady(repository, sourcev1.GitOperationFailedReason, err.Error()), err
 	}
-
-	// return early on unchanged revision
 	artifact := r.Storage.NewArtifactFor(repository.Kind, repository.GetObjectMeta(), revision, fmt.Sprintf("%s.tar.gz", commit.Hash()))
-	if apimeta.IsStatusConditionTrue(repository.Status.Conditions, meta.ReadyCondition) && repository.GetArtifact().HasRevision(artifact.Revision) {
+
+	// copy all included repository into the artifact
+	includeArtifcats := []*sourcev1.Artifact{}
+	for _, incl := range repository.Spec.Include {
+		dName := types.NamespacedName{Name: incl.GitRepository.Name, Namespace: incl.GitRepository.Namespace}
+		var gr sourcev1.GitRepository
+		err := r.Get(context.Background(), dName, &gr)
+		if err != nil {
+			return sourcev1.GitRepositoryNotReady(repository, meta.DependencyNotReadyReason, err.Error()), err
+		}
+		includeArtifcats = append(includeArtifcats, gr.GetArtifact())
+
+		err = r.Storage.CopyToPath(gr.GetArtifact(), filepath.Join(tmpGit, incl.Path))
```

Comment:
Should this be moved further down so that files are not copied when the repository returns early?

Reply:
I think the whole `for` block would need to be moved under the if condition below.

Reply:
Yep you are right as it would reduce io operations that will be discarded anyways.